### ISSUE TITLE
Angular 12.2.2 css minimizer build error - Invalid version: "TP"

### ISSUE THREAD

**Issue by geo242** [angular/angular-cli#21699]
# 🐞 Bug report

- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

Worked in 12.2.1 and broken in >=12.2.2

### Description

Build error during css minification.

## 🔬 Minimal Reproduction

`ng build` 

optimization.styles.minify must be true in the angular.json

May require scss styles

## 🔥 Exception or Error

This error shows for every component in the build

<pre><code>
some-path/some-component.component.scss from Css Minimizer
Error: Transform failed with 1 error:
error: Invalid version: "TP"
    at failureErrorWithLog (/home/geoff/code/iverticle/web-apps/admin/node_modules/esbuild/lib/main.js:1449:15)
    at /home/geoff/code/iverticle/web-apps/admin/node_modules/esbuild/lib/main.js:1260:29
    at /home/geoff/code/iverticle/web-apps/admin/node_modules/esbuild/lib/main.js:609:9
    at handleIncomingPacket (/home/geoff/code/iverticle/web-apps/admin/node_modules/esbuild/lib/main.js:706:9)
    at Socket.readFromStdout (/home/geoff/code/iverticle/web-apps/admin/node_modules/esbuild/lib/main.js:576:7)
    at Socket.emit (events.js:315:20)
    at addChunk (internal/streams/readable.js:309:12)
    at readableAddChunk (internal/streams/readable.js:284:9)
    at Socket.Readable.push (internal/streams/readable.js:223:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:188:23)
</code></pre>

## 🌍 Your Environment

<pre><code>
Angular CLI: 12.2.2
Node: 14.16.1
Package Manager: npm 6.14.12
OS: linux x64

Angular: 12.2.2
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1202.2
@angular-devkit/build-angular   12.2.2
@angular-devkit/core            12.2.2
@angular-devkit/schematics      12.2.2
@angular/cdk                    12.2.4
@angular/flex-layout            12.0.0-beta.34
@schematics/angular             12.2.2
rxjs                            6.6.7
typescript                      4.2.4
    

</code></pre>

**Anything else relevant?**

N/A

**Comment by alan-agius4** [comment]
This seems like a bug but we'll need to look at a reproduction to find and fix the problem. Can you setup a minimal repro please?

You can read [here](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-submitting-an-issue) why this is needed. A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

This might be related to your directory structure so its really important to get an accurate repro to diagnose this.

**Comment by alan-agius4** [comment]
Can you provide the browserslist configuration file?

**Comment by geo242** [comment]
@alan-agius4 this is the entire file...

<pre><code>
# This file is used by the build system to adjust CSS and JS output to support the specified browsers below.
# For additional information regarding the format and rule options, please see:
# https://github.com/browserslist/browserslist#queries

# You can see what browsers were selected by your queries by running:
#   npx browserslist

supports es6-module
</code></pre>

**Comment by alan-agius4** [comment]
The above query will yield a large set of browsers which most of them are unsupported by the Angular framework

```
and_chr 92
and_ff 90
and_qq 10.4
android 92
chrome 95
chrome 94
chrome 93
chrome 92
chrome 91
chrome 90
chrome 89
chrome 88
chrome 87
chrome 86
chrome 85
chrome 84
chrome 83
chrome 81
chrome 80
chrome 79
chrome 78
chrome 77
chrome 76
chrome 75
chrome 74
chrome 73
chrome 72
chrome 71
chrome 70
chrome 69
chrome 68
chrome 67
chrome 66
chrome 65
chrome 64
chrome 63
chrome 62
chrome 61
edge 92
edge 91
edge 90
edge 89
edge 88
edge 87
edge 86
edge 85
edge 84
edge 83
edge 81
edge 80
edge 79
edge 18
edge 17
edge 16
firefox 93
firefox 92
firefox 91
firefox 90
firefox 89
firefox 88
firefox 87
firefox 86
firefox 85
firefox 84
firefox 83
firefox 82
firefox 81
firefox 80
firefox 79
firefox 78
firefox 77
firefox 76
firefox 75
firefox 74
firefox 73
firefox 72
firefox 71
firefox 70
firefox 69
firefox 68
firefox 67
firefox 66
firefox 65
firefox 64
firefox 63
firefox 62
firefox 61
firefox 60
ios_saf 14.5-14.7
ios_saf 14.0-14.4
ios_saf 13.4-13.7
ios_saf 13.3
ios_saf 13.2
ios_saf 13.0-13.1
ios_saf 12.2-12.4
ios_saf 12.0-12.1
ios_saf 11.3-11.4
ios_saf 11.0-11.2
ios_saf 10.3
op_mob 64
opera 78
opera 77
opera 76
opera 75
opera 74
opera 73
opera 72
opera 71
opera 70
opera 69
opera 68
opera 67
opera 66
opera 65
opera 64
opera 63
opera 62
opera 60
opera 58
opera 57
opera 56
opera 55
opera 54
opera 53
opera 52
opera 51
opera 50
opera 49
opera 48
safari 15
safari 14.1
safari 14
safari 13.1
safari 13
safari 12.1
safari 12
safari 11.1
safari 11
safari 10.1
safari TP
samsung 14.0
samsung 13.0
samsung 12.0
samsung 11.1-11.2
samsung 10.1
samsung 9.2
samsung 8.2
```


I suggest that you update the browserslist to include only the browsers that are supported.
```
# This file is used by the build system to adjust CSS and JS output to support the specified browsers below.
# For additional information regarding the format and rule options, please see:
# https://github.com/browserslist/browserslist#queries

# For the full list of supported browsers by the Angular framework, please see:
# https://angular.io/guide/browser-support

# You can see what browsers were selected by your queries by running:
#   npx browserslist

last 1 Chrome version
last 1 Firefox version
last 2 Edge major versions
last 2 Safari major versions
last 2 iOS major versions
Firefox ESR
```

Which yields the following
```
hrome 92
edge 92
edge 91
firefox 91
firefox 78
ios_saf 14.5-14.7
ios_saf 14.0-14.4
ios_saf 13.4-13.7
ios_saf 13.3
ios_saf 13.2
ios_saf 13.0-13.1
safari 14.1
safari 14
safari 13.1
safari 13
```

For a full list of supported browsers please see https://angular.io/guide/browser-support

**Comment by geo242** [comment]
Thank you for that @alan-agius4 

Without changing the browserslist file, I have been doing a lot of testing by changing the node and npm versions and removing the node_modules folder and the package-lock and reinstalling the dependencies and re-running the build. This is how I narrowed it down to 12.2.1 to 12.2.2. However, since putting in the issue, I have continued to to do this and at some point the problem has just miraculously disappeared and I can't reproduce it anymore. 

I don't think this had anything to do with the browserslist, but rather something to do with my environment which seems impossible to replicate now. So, thank you for your help, but I am closing the issue.
